reactor-netty-core/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java

```diff
@@ -471,7 +484,7 @@ ConnectionObserver owner() {
 				obs = channel.attr(OWNER)
 				             .get();
 				if (obs == null) {
-					obs = new PendingConnectionObserver();
+					obs = new PendingConnectionObserver(context);
```

Comment:
I prefer this context to be available only when establishing a connection.
Basically we do not have OWNER (or PendingConnectionObserver can appear) in two cases:
- when establishing a connection
- when returning a connection to the pool it is removed and once the connection is acquired it is set again

Can we add something like the code below after this row
https://github.com/reactor/reactor-netty/blob/631f7a67269ac450195d2336714edb9098e6736f/reactor-netty-core/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java#L533

```
channel.attr(OWNER).compareAndSet(null, obs)
```